framework/src/main/java/org/checkerframework/framework/stub/AnnotationFileParser.java

```diff
@@ -343,16 +343,44 @@ public RecordStub(LinkedHashMap<String, RecordComponentStub> componentsByName) {
     /** The type of the record component. */
     public final AnnotatedTypeMirror type;
 
+    /**
+     * The set of all annotations on the declaration of the record component. If applicable these
+     * will be copied to the corresponding field, accessor method, and compact canonical constructor
+     * parameter.
+     */
+    private final Set<AnnotationMirror> allAnnotations;
+
     /** Whether this component has an accessor of exactly the same name in the stubs file. */
     private boolean hasAccessorInStubs = false;
 
     /**
      * Creates a new RecordComponentStub with the given type.
      *
      * @param type the type of the record component
+     * @param allAnnotations the declaration annotations on the component
      */
-    public RecordComponentStub(AnnotatedTypeMirror type) {
+    public RecordComponentStub(AnnotatedTypeMirror type, Set<AnnotationMirror> allAnnotations) {
       this.type = type;
+      this.allAnnotations = allAnnotations;
+    }
+
+    /**
+     * Get the record component annotations that are applicable to the given element kind.
+     *
+     * @param elementKind the element kind to apply to (e.g., FIELD, METHOD)
+     * @return the set of annotations from the component that apply
+     */
+    public Set<AnnotationMirror> getAnnotationsForTarget(ElementKind elementKind) {
+      HashSet<AnnotationMirror> filtered = new HashSet<>();
```

Comment:
This use of `HashSet` makes this method nondeterministic.  Below, `AnnotationUtils.createAnnotationSet` returns a deterministic set (in particular, a `TreeSet`), and I think retaining that determinism would be best.

Reply:
Done.